### ISSUE TITLE
imagePuller PODs failing due to OOM

### ISSUE THREAD

**Issue by rafaeltuelho** [eclipse/che#22446]
### Describe the bug

ImagePuller PODs failing due to OOM

### Che version

7.72@latest

### Steps to reproduce

1. Create a `CheCluster` resource with imagePuller enabled
for instance:

```yaml
    imagePuller:
      enable: true
      spec:
        images: >-
          universal-developer-image-ubi8-latest=quay.io/redhat_na_ssa/universal-developer-image@sha256:72b8edb995f76eb5575e7c3d01c787abed78023ea96d12c0ef54493971c7e299;universal-developer-image-ubi8-latest=quay.io/devfile/universal-developer-image@sha256:9777fcda747644a69c5422013a263413dd1268d55b04a5d5195f52ff881af107
```

3. During provisioning imagePuller PODs start failing due to insufficient memory
POD's events
```
Error: container create failed: time="2023-08-21T17:34:41Z" level=error msg="runc create failed: unable to start container process: container init was OOM-killed (memory limit too low?)"
```

### Expected behavior

ImagePuller PODs should have enough resources to complete its designated task by default.

### Runtime

OpenShift

### Screenshots

![image](https://github.com/eclipse/che/assets/39063/bcacf34d-858e-4c47-b28b-d5a1dadc60ce)

### Installation method

OperatorHub

### Environment

Linux

**Comment by tolusha** [comment]
I could see such a message, but after a couple of seconds pod started successfully.

**Comment by rafaeltuelho** [comment]
@tolusha in my cluster, the POD keeps restarting forever. Maybe it depends on the image size is being downloaded.

**Comment by tolusha** [comment]
Could you try if it solves your issue:
`oc patch checluster eclipse-che --patch '{"spec": {"components": {"imagePuller": {"spec" : {"cachingMemoryLimit": "30Mi"}}}}}' --type=merge -n eclipse-che`

**Comment by rafaeltuelho** [comment]
Yeah, that solves the problem. Tha was my workaround.
My ask with this issue here is to adjust the initial default limit set by the operator so this problem doesn't happen with the default values.

**Comment by tolusha** [comment]
Fixed of Kubernetes Image Puller Operator 1.0.6
